src/transformStateWithClones.js

```diff
@@ -8,6 +8,34 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateClone = { ...state };
+  const stateArr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ('addProperties'):
+        Object.assign(stateClone, action.extraData);
+        break;
+
+      case ('removeProperties'):
+        for (const keyToRemove of action.keysToRemove) {
+          delete stateClone[keyToRemove];
+        }
+        break;
+
+      case ('clear'):
+        for (const key in stateClone) {
+          delete stateClone[key];
+        }
+        break;
+
+      default:
+        throw new Error();
```

Comment:
I don't know for sure but I think it's better to give a descriptive text to the Error.)